unicode.cpp

```diff
@@ -224,138 +224,244 @@ static std::vector<size_t> unicode_regex_split_custom_gpt2(const std::string & t
     std::vector<size_t> bpe_offsets; // store the offset of each word
     bpe_offsets.reserve(offsets.size()); // Reserve memory for the approximate size
 
-    size_t start = 0;
-
     const auto cpts = unicode_cpts_from_utf8(text);
```

Comment:
perf note: this call can be avoided since we already computed the codepoints in `unicode_regex_split()`. We just need to pass a const reference in the custom functions

Reply:
Yup, and `text_collapsed` should be computed lazily (thx @ibob for the hint). But we can do all these optimizations after we merge this. For let's focus on verifying the correctness with some large diverse corpus